fms/utils/serialization.py

```diff
@@ -86,8 +91,30 @@ def get_adapted(
 from fms import models
 
 
+def get_ckp_format(model_path: Union[str, Path]) -> str:
+    """
+    Returns the checkpoint format of a model checkpoint. If format
+    is not supported, returns "unk"
+
+    Args:
+    model_path: the path to find the weights.
+    """
+    model_path = Path(os.path.expanduser(model_path))
+    if model_path.suffix == ".pth" or len(sorted(model_path.glob("*.pth"))) > 0:
+        return "pt"
+    if model_path.suffix == ".bin" or len(sorted(model_path.glob("*.bin"))) > 0:
+        return "hf"
+    if (
+        model_path.suffix == ".safetensors"
+        or len(sorted(model_path.glob("*.safetensors"))) > 0
+    ):
+        return "st"
+    return "unk"
```

Comment:
since we don't have any handling for 'unk' it seems we should do somethign else here. E.g. we could assume files with an unrecognized extension must by pytorch checkpoints (return pt by default) or we could raise an error if we think that's unlikely. I suspect we'll often enough encounter checkpoints like .ckp or something that would lead to a 'unk' here but are valid pytorch checkpoints.

Reply:
also, I wonder if we should favor safetensors when they're available

Reply:
returning "pt" by default now and moving the safetensors check first